pkg/asset/cluster/tfvars.go

```diff
@@ -293,11 +293,11 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			workerConfigs[i] = w.Spec.Template.Spec.ProviderSpec.Value.Object.(*gcpprovider.GCPMachineProviderSpec)
 		}
 		if installConfig.Config.Publish == types.ExternalPublishingStrategy {
-			publicZone, err := gcpconfig.GetPublicZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.BaseDomain)
+			_, err := gcpconfig.GetPublicZoneByID(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.GCP.PublicZoneID)
 			if err != nil {
 				return errors.Wrapf(err, "failed to get GCP public zone")
 			}
-			publicZoneName = publicZone.Name
+			publicZoneName = installConfig.Config.GCP.PublicZoneID
```

Comment:
in don't understand what this change is doing? it is requesting for a zone id using api and continuing to use installconfig set value?

Reply:
this should probably be https://github.com/openshift/installer/pull/2764/commits/b5653abad35955e4661656d8b80f3b2750d4db61#diff-2270a1f17144108c0e530a04c17900c4R108-R116

Reply:
Fixed.